src/main/java/org/craftercms/core/service/Context.java

```diff
@@ -51,4 +53,10 @@ public interface Context extends Cloneable {
 
     Context clone();
 
+    /**
+     * Config variables for the context. e.g.: ${siteName} will be replaced by the value of the siteName variable
+     *
+     * @return Return a map of variables to be used when loading configuration files.
+     */
+    Map<String, String> getConfigLookupVariables();
```

Comment:
Maybe we should call this something like attributes and make it more general

Reply:
I'm ok with this name `configLookupVariables`